### DISCUSSION TITLE
6. Projedeki insertXOrdered Metodlarinin Erisim Belirleyicilerinin Degistirilmesi

### DISCUSSION THREAD

**Discussion by gusanmaz** [gusanmaz/NKU_DS_Course_2022#17] [Announcements]
6. Projedeki insertXOrdered metodlarinin (`insertNameOrdered`, `insertSurnameOrdered`, `insertIDOrdered` ve `insertGPAOrdered` erisim belirleyicileri public'den private'e donusturuldu. Eger proje davet linkini 22 Mayis 2022 17:30 oncesi tiklayip repo olusturduysaniz kodunuzdaki bu degisiklikleri manuel olarak yapmaniz gerekmektedir.

Proje davet linkini belirtilen tarihten sonra kabul eden arkadaslarin hazir proje kodlarinda bir degisiklik yapmasina gerek yoktur cunku bu metodlar sizin repolarinizda repo ilk olusturuldugunda private olarak gozukecek.

**Sizce bu metodlarin erisim belirleyicilerini neden private olarak degistirme geregi duymus olabilirim?** Bu soruya bu baslik altinda verilecek dogru yanitlar arasindaki en kapsamli 3 yanit odev puanina ekstra puan olarak yansitilacaktir!

**Comment by gusanmaz** [comment]
Sadece 24 Mayis 2022 12:30'dan once yapilan yorumlar degerlendirmeye alinacaktir.

**Comment by yazilimdx** [comment]
BST clasina  ait  property lerden olan keyType ozelliginin  alacagi  degerler uzerine ikili arama agac yapisi
olusturmak istedik ve bu nedenle ikili arama agac nesnesi  tanimlandigi anda  ikili arama agac nesnesinin  referans  alacagi
{ID,name,surname,GPA} ozelliklerinden  hangisi  oludugunu  belirlemis olduk .

Ve bu yuzden  bir  BST adli class  nesnesine  yani  ikili arama agac yapisina  biz  sonradan kafamiza  gore  
ekleme  bicimi  gerceklestiremeyelim diye de (insertNameOrdered, insertSurnameOrdered, insertIDOrdered ve insertGPAOrdered)
metodlarinin erisim  belirteclerini  "private" olarak  belirledik ve  "insert" adli  fonksiyona  "keyType" seceneklerine  gore  if  kosullari  
olusturarak  sadece  burdan ulasilabilmesini saglamis olduk .

**Reply by gusanmaz** [reply]
Aciklamadaki agac ifadeleri ikili arama agaci olarak degistirilirse daha dogru olur.

**Reply by yazilimdx** [reply]
Degistirdim hocam .

**Comment by umutcanclkk** [comment]
BST  de  bulunan  keyType  ozelliginin degerine  gore  agac  yapisi oluşturulduğundan , insert adli  metod  ile  seçilen  keyType  disinda  ekleme yapmasini istemediğimiz icin  erisim belirteçlerini  private  yaptik .
